batch_job/src/batch_task.c

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright (C) 2018- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include "batch_task.h"
+#include "stringtools.h"
+#include "xxmalloc.h"
+
+struct batch_task *batch_task_create(struct batch_queue *queue)
+{
+	struct batch_task *t = malloc(sizeof(struct batch_task));
```

Comment:
Given the 0 and NULL to follow (and on fields that may be added later), I suggest:

struct batch_task *t = calloc(1,sizeof(*t));

Then you can remove the = 0 and = NULL.